sematic/resolvers/state_machine_resolver.py

```diff
@@ -92,6 +99,17 @@ def _run_inline(self, future: AbstractFuture):
     def _wait_for_scheduled_run(self) -> None:
         pass
 
+    def _handle_sigint_cancel(self, signum, frame):
+        logger.info("Received SIGINT, canceling resolution...")
+        self._cancel_futures()
+        self._resolution_did_cancel()
+        exit(1)
```

Comment:
Note that `_cancel_futures` does not actually update anything in the DB, it simply sets the state of in-memory futures to `CANCELED` and is idempotent.

Reply:
I believe my other points stand.

Reply:
Reading this does also remind me that we are wanting to move to a place where the DB + server is the source of truth for status information. So rather than the resolver setting the futures to CANCELED, because it assumes the server will cancel them, it should ideally:
(a) notify the server of the intent to cancel
(b) poll the server for updates on the statuses of the runs and set them to CANCELED as the server reports them as such

Alternatively:
(a) notify the server of the intent to cancel
(b) do one update of the run statuses (maybe this isn't even necessary)
(c) just exit. The server will update the statuses of the child runs as it cancels them

Reply:
Yes, I agree with points (a) and (c) from the alternative version.

Reply:
The way it currently works:

Cancelation is triggered from the CLI:
- resolver `PUT`s to server to cancel.
- server cancels resolution and ALL unfinished runs, publishes Socket IO message
- resolver receives Socket IO message, sets all unfinished futures to CANCELED (runs have already been updated), and exits

Cancelation triggered from the UI:
- server cancels resolution and ALL unfinished runs, publishes Socket IO message
- resolver receives Socket IO message, sets all unfinished futures to CANCELED (runs have already been updated), and exits

arguable the last part (settings unfinished futures to CANCELED) is not necessary since we are exiting.

The server + DB is still the source of truth for cancelation, we are only updating local futures to clean up. I am happy to remove `_cancel_futures` since it is actually not necessary I believe.

Reply:
I vote we remove `_cancel_futures` and leave the rest of the behavior as-is

Reply:
Yes, please do remove it; it would simplify the flow.